### ISSUE TITLE
Differential privacy via opacus

### ISSUE THREAD

**Issue by minseok-ryu** [APPFL/APPFL#349]
### How did you install appfl?

```console
install from source
```

### What version of appfl are you using?

v1.8.0

### Describe the problem.

I am running "run_serial.py" with differential privacy through Opacus. 

Dataset: MNIST
Model: CNN

1. The current cnn model should be revised to compute per-sample gradient through Opacus. Specifically, line 14 we should have "self.act = nn.ReLU(inplace=False)"
2. It is worth-noting that resnet models with "batch normalization" layers will not work as indicated in [Link](https://opacus.ai/tutorials/building_image_classifier)
3.  Opacus does per-sample gradient clipping. Therefore, you do not need:

```
if getattr(self.train_configs, "clip_grad", False) or getattr(
            self.train_configs, "use_dp", False
        ):
            assert hasattr(self.train_configs, "clip_value"), (
                "Gradient clipping value must be specified"
            )
            assert hasattr(self.train_configs, "clip_norm"), (
                "Gradient clipping norm must be specified"
            )
            torch.nn.utils.clip_grad_norm_(
                self.model.parameters(),
                self.train_configs.clip_value,
                norm_type=self.train_configs.clip_norm,
            )
```

This clips the `mini-batch gradient`, but in DP, we need to clip the `per-sample gradient`. If you need to clip `mini-batch gradient` for some reason other then DP, you can revise the code as:

```
if getattr(self.train_configs, "clip_grad", False):
            assert hasattr(self.train_configs, "clip_value"), (
                "Gradient clipping value must be specified"
            )
            assert hasattr(self.train_configs, "clip_norm"), (
                "Gradient clipping norm must be specified"
            )
            torch.nn.utils.clip_grad_norm_(
                self.model.parameters(),
                self.train_configs.clip_value,
                norm_type=self.train_configs.clip_norm,
            )
```

4. Since Opacus is already used, we may consider removing `Laplace` and `Gaussian` mechanisms, implemented long ago. 

5. Since Opacus is used, we may consider incorporating the RDP accountant to compute the privacy loss.
